vnext/Microsoft.ReactNative/Fabric/Composition/ReactNativeIsland.cpp

```diff
@@ -903,6 +932,16 @@ void ReactNativeIsland::OnUnmounted() noexcept {
   if (!m_mounted)
     return;
   m_mounted = false;
+
+  if (m_island && m_island.IsConnected()) {
+    auto focusController = winrt::Microsoft::UI::Input::InputFocusController::GetForIsland(m_island);
+    auto request = winrt::Microsoft::UI::Input::FocusNavigationRequest::Create(
+        winrt::Microsoft::UI::Input::FocusNavigationReason::Programmatic);
+    if (focusController.HasFocus()) {
+      focusController.DepartFocus(request);
+    }
+  }
+
   if (auto componentView = GetComponentView()) {
```

Comment:
Adding a check here for !m_isFragment will fix Modal in UseExperimentalWinUI3 flag.

Currently Modal (can see this on the second Modal in rntester) will error out in UseExperimentalWinUI3 when closed because the component has already been removed from ViewRegistry